dot/config_test.go

```diff
@@ -0,0 +1,496 @@
+		Enabled        bool
+		External       bool
+		Unsafe         bool
+		UnsafeExternal bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "default",
+			want: false,
+		},
+		{
+			name:   "enabled true",
+			fields: fields{Enabled: true},
+			want:   true,
+		},
+		{
+			name:   "external true",
+			fields: fields{External: true},
+			want:   true,
+		},
+		{
+			name:   "unsafe true",
+			fields: fields{Unsafe: true},
+			want:   true,
+		},
+		{
+			name:   "unsafe external true",
+			fields: fields{UnsafeExternal: true},
+			want:   true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			r := &RPCConfig{
+				Enabled:        tt.fields.Enabled,
+				External:       tt.fields.External,
+				Unsafe:         tt.fields.Unsafe,
+				UnsafeExternal: tt.fields.UnsafeExternal,
+			}
+			got := r.isRPCEnabled()
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRPCConfig_isWSEnabled(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		WS               bool
+		WSExternal       bool
+		WSUnsafe         bool
+		WSUnsafeExternal bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "default",
+			want: false,
+		},
+		{
+			name:   "ws true",
+			fields: fields{WS: true},
+			want:   true,
+		},
+		{
+			name:   "ws external true",
+			fields: fields{WSExternal: true},
+			want:   true,
+		},
+		{
+			name:   "ws unsafe true",
+			fields: fields{WSUnsafe: true},
+			want:   true,
+		},
+		{
+			name:   "ws unsafe external true",
+			fields: fields{WSUnsafeExternal: true},
+			want:   true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			r := &RPCConfig{
+				WS:               tt.fields.WS,
+				WSExternal:       tt.fields.WSExternal,
+				WSUnsafe:         tt.fields.WSUnsafe,
+				WSUnsafeExternal: tt.fields.WSUnsafeExternal,
+			}
+			got := r.isWSEnabled()
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_networkServiceEnabled(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		cfg *Config
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "dev config",
+			args: args{cfg: DevConfig()},
+			want: true,
+		},
+		{
+			name: "empty config",
+			args: args{cfg: &Config{}},
+			want: false,
+		},
+		{
+			name: "core roles 0",
+			args: args{cfg: &Config{
+				Core: CoreConfig{
+					Roles: 0,
+				},
+			}},
+			want: false,
+		},
+		{
+			name: "core roles 1",
+			args: args{cfg: &Config{
+				Core: CoreConfig{
+					Roles: 1,
+				},
+			}},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got := networkServiceEnabled(tt.args.cfg)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRPCConfig_String(t *testing.T) {
+	tests := []struct {
+		name   string
+		fields RPCConfig
```

Comment:
rename `fields` to `config` / `rpcConfig`

Reply:
updated to `rpcConfig`